### DISCUSSION TITLE
OAuth requests not working locally

### DISCUSSION THREAD

**Discussion by Fynkymynky** [ddev/ddev#4514] [Q&A] [answered]
I'm trying to make a request to a local API which uses three legged OAuth. But my scripts don't work locally.

```
    $this->oauth = new \OAuth($this->API_CONSUMER_KEY, $this->API_CONSUMER_SECRET, OAUTH_SIG_METHOD_HMACSHA1, OAUTH_AUTH_TYPE_URI);
    $this->oauth->setToken($this->API_TOKEN, $this->API_TOKEN_SECRET);
    $this->oauth->fetch($this->domain . '/rest/v1/' . $query);
    $json = $this->oauth->getLastResponse();
```

I've just started using ddev coming from a Vagrant setup with no code changes. The script works fine calling the production server but not locally. When I disable the authentication requirements within Drupal the script works fine.

I can't work out why the the authentication won't work with ddev?

The following error is reported when running the script:

```
OAuthException Object
(
    [message:protected] => Invalid auth/bad request (got a 401, expected HTTP/1.1 20X or a redirect)
    [string:Exception:private] => 
    [code:protected] => 401
    [file:protected] => /var/www/html/scripts/api/test.php
    [line:protected] => 37
    [trace:Exception:private] => Array
        (
            ....
        )

    [previous:Exception:private] => 
    [lastResponse] => ["Invalid signature"]
    [debugInfo] => 
)
```

I'm running PHP8 with php8.0-oauth installed.

**Comment by rfay** [comment]
Where are you running the script?

Can you simplify it? 

What hostname is the script trying to hit?

**Reply by Fynkymynky** [reply]
Hi @rfay, thank you for the response. I changed my web server from `nginx` to `apache` and the script now works. I'm not sure why but PHP-Oauth doesn't seem to play nice with `nginx`.

I still have to run the full test suite, but the simple script works. thanks again.

**Comment by Fynkymynky** [comment, marked as answer]
The one thing I didn't check was what type of web server was running. I had left it as default `nginx-fpm` but once I changed it to `apache-fpm` the script now works!

I don't know why `nginx` doesn't work? But I can stick with `apache` locally.

**Reply by rfay** [reply]
Glad you got it working. Perhaps your recipe depends on a .htaccess file?

**Reply by Fynkymynky** [reply]
oh! maybe? good idea, I'll check that out. Thank you.
